piet-cairo/src/lib.rs

```diff
@@ -165,6 +171,20 @@ impl<'a> RenderContext for CairoRenderContext<'a> {
         self.ctx.move_to(pos.x, pos.y);
         self.ctx.show_text(&layout.text);
     }
+
+    fn save(&mut self) {
+        self.ctx.save();
+    }
+
+    fn restore(&mut self) {
+        self.ctx.restore();
+    }
+
+    fn finish(&mut self) {}
```

Comment:
Will the cairo and web versions of finish always be empty? I worry about learnability and useability when forgetting to call `finish` on web and cairo are inconsequential but could break things on direct-2d

Reply:
> a utility function that finishes the render context right before extracting the bitmap

This sounds like https://www.cairographics.org/manual/cairo-cairo-surface-t.html#cairo-surface-flush, but there’s also something named `cairo_surface_finish`.